cmake/CMakeLists.txt

```diff
@@ -880,8 +880,8 @@ else()
         if(${onnxruntime_target_platform} MATCHES "^(ARM.*|arm.*)$" )
             message(WARNING "Cpuinfo not included for compilation problems with Windows ARM.")
             set(CPUINFO_SUPPORTED FALSE)
-        elseif(WINDOWS_STORE)
-            message(WARNING "Cpuinfo not included in Windows Store builds")
+        elseif(WINDOWS_STORE OR (WIN32 AND NOT CMAKE_CXX_STANDARD_LIBRARIES MATCHES kernel32.lib))
```

Comment:
Can we add this check into a varilable, and use it rather than specifying it explicitly? It is a little esoteric as it is...

Reply:
But, what about normal Win32 build? Does the generated binary require Win8+ ? If yes, we should disable CPUINFO too.

Reply:
If it's linking to desktop libraries it should be fine I believe. GetMaximumProcessorCount, for example, is available since Windows 7. The issue is WCOS and UWP, which have a limited API.

Reply:
Which team is in charge of validating ORT builds without WinML (e.g. no_contrib_ops) on Windows 8, ORT or WinML? If we're expected to validate it I can ping my team